### ISSUE TITLE
Transition from leader to follower sometime fails

### ISSUE THREAD

**Issue by deepthidevaki** [zeebe-io/zeebe#5291]
**Describe the bug**
In one cluster we observed the following:

Zeebe-0 was the leader for partition 1. Then there was a new leader. Zeebe-0 closes its leader services. Partition is them marked as unhealthy and it remains unhealthy for ever. It seems then no follower services are installed. As a result it didnot receive any new snapshots, but was still replicating log. So it could not compact the logs and eventually the node went out of disk space.

**Environment:**
 - Zeebe Version: stable/0.23

**Comment by npepinpe** [comment]
I think this will be fixed by #5522 ?

**Comment by MiguelPires** [comment]
The refactor won't fix it but the fix for #5117 will I think. The refactor just makes it possible to write tests for it so it needs to be merged before I can work on the bug
